src/services_models/reserves.jl

```diff
@@ -150,23 +166,21 @@ function cost_function!(
     return
 end
 
-function service_requirement_constraint!(
+function add_constraints!(
     container::OptimizationContainer,
+    T::Type{RequirementConstraint},
     service::SR,
-    ::ServiceModel{SR, StepwiseCostReserve},
-) where {SR <: PSY.ReserveDemandCurve}
+    contributing_devices::U,
+    model::ServiceModel{SR, StepwiseCostReserve},
+) where {
+    SR <: PSY.ReserveDemandCurve,
+    U <: Union{Vector{D}, IS.FlattenIteratorWrapper{D}},
+} where {D <: PSY.Component}
     initial_time = get_initial_time(container)
     @debug initial_time
     time_steps = get_time_steps(container)
     name = PSY.get_name(service)
```

Comment:
```suggestion
   names = [PSY.get_name(s) for s in services]
```

Please avoid patterns like creating single element arrays like in line 183. If a user uses a multi-area model this will fail.

Reply:
I'm not sure if we actually support multi-area models? Here we just have one service so the change you recommend will not work unless I do make a single element array.
`names = [PSY.get_name(s) for s in [service]]`